guides/dsl.md

```diff
@@ -0,0 +1,289 @@
+# Eve JavaScript/TypeScript DSL
+
+The forthcoming v0.3 release of Eve supports a domain specific langauge (DSL) that interacts with the Eve server, allowing you to intermix Javascript and Eve code. There are several benefits to using the DSL:
+
+1) The syntax is native Javascript, so if you know JS you can write in the DSL
+2) You can interact with Javascript functions and libraries
+3) The DSL can be used "a la carte", so you can use as much of Eve as you need to for your project. Thus you can write an entire application in Eve, just use it as a datastore, or anywhere between. This makes it easy to integrate Eve with an existing Javascript application.
+4) You can easily import data into Eve. If you can get your data in Javascript, you can use it in Eve.
+
+## Using the DSL
+
```

Comment:
We need a little preface here saying that the guide is written using ES6. Specifically, we use and recommend TypeScript ^2.1.0 (which transpiles ES6 to cross-platform compatible javascript. Specifically, for concision and readability we make frequent use of destructuring [1] and arrow functions [2].

[1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
[2] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions#Arrow_functions